packages/ethereum-plugin/src/validiators/toEmit.ts

```diff
@@ -0,0 +1,60 @@
+import { Control, Expectation, getControl, isEqual } from 'earljs/internals'
+import { Contract, providers } from 'ethers'
+import { LogDescription } from 'ethers/lib/utils'
+
+export async function toEmit(this: Expectation<any>, contract: Contract, event: string, args?: any[]): Promise<void> {
```

Comment:
Let's rethink this interface so it can be made typesafe ie. if the contract type comes from TypeChain (extends `BaseContract`) IDE should prompt possible events, args etc . I think the very same signature could be made typesafe with a generic argument `extends BaseContract`. But please ensure that all of this also works for non-typechain users. 

I wonder should we opt out from using arrays to match args and prefer object (when event has named args, which is often a case...). Thoughts?